### ISSUE TITLE
No source code was seen during the build with ninja

### ISSUE THREAD

**Issue by Manouchehri** [github/codeql-cli-binaries#21]
```sh
cd /mnt/builds/chromium/src/
gn gen out/linux-x64-debug --args='target_os="linux" target_cpu="x64" is_debug=true is_component_build=true symbol_level=2 ffmpeg_branding="Chromium" proprietary_codecs=false android_full_debug=true blink_symbol_level=2 dcheck_is_configurable=true enable_mojom_message_id_scrambling=false enable_profiling=true v8_enable_backtrace=true v8_enable_disassembler=true v8_enable_object_print=true v8_expose_symbols=true v8_postmortem_support=true enable_nacl=false v8_no_inline=true' --export-compile-commands
/mnt/builds/codeql/codeql database create codeqldb --command="ninja -C out/linux-x64-debug d8" --language=cpp --threads=11
...
[2020-03-18 14:23:33] [build] [1422/1422] LINK ./d8
Finalizing database at /mnt/builds/chromium/src/codeqldb.
A fatal error occurred: No source code was seen during the build.
This can occur if the specified build commands failed to compile or process any code.
 - Confirm that there is some source code for the specified language in the project.
 - For codebases written in Go, JavaScript, TypeScript, and Python, do not specify 
   an explicit --command.
 - For other languages, the --command must specify a "clean" build which compiles 
   all the source code files without reusing existing build artefacts.
```

I've also tried indexing everything with the commands below, which seems to ignore nearly everything (only 314 files indexed).

```sh
cd /mnt/builds/chromium/src/
gn gen out/linux-x64-debug --args='target_os="linux" target_cpu="x64" is_debug=true is_component_build=true symbol_level=2 ffmpeg_branding="Chromium" proprietary_codecs=false android_full_debug=true blink_symbol_level=2 dcheck_is_configurable=true enable_mojom_message_id_scrambling=false enable_profiling=true v8_enable_backtrace=true v8_enable_disassembler=true v8_enable_object_print=true v8_expose_symbols=true v8_postmortem_support=true enable_nacl=false v8_no_inline=true' --export-compile-commands
cd out/linux-x64-debug
/mnt/builds/codeql/codeql database create codeql --language=cpp --threads=11
```

**Comment by Manouchehri** [comment]
Hmm, should the extractor actually be exiting out here on something like `d8.cc`?

```
[T 14:19:37 29432] Attempting to switch stdout/stderr to 4...
Testing stderr (2/4).
[T 14:19:37 29432] Recognising file, first four bytes: { 7f, 45, 4c, 46 }
[T 14:19:37 29432]   ELF 64-bit x86-64.
[T 14:19:37 29432]   PH#1: PT_DYNAMIC or PT_INTERP - so dynamic.
[T 14:19:37 29431] Reading configuration file /mnt/builds/chromium/src/codeqldb/working/tracing/compiler-tracing11526775434541759650.spec...
[T 14:19:37 29431] ==== Candidate to intercept: ../../third_party/llvm-build/Release+Asserts/bin/clang++ (canonical: /mnt/builds/chromium/src/third_party/llvm-build/Release+Asserts/bin/clang) ====
[T 14:19:37 29431] === Intercepted call to /mnt/builds/chromium/src/third_party/llvm-build/Release+Asserts/bin/clang === 
[T 14:19:37 29432] Reading configuration file /mnt/builds/chromium/src/codeqldb/working/tracing/compiler-tracing11526775434541759650.spec...
[E 14:19:38 29432] CodeQL C/C++ Extractor 1.24.0-SNAPSHOT
[E 14:19:38 29432] Current directory: /mnt/builds/chromium/src/out/linux-x64-debug
[E 14:19:38 29432] Command: /mnt/builds/codeql/cpp/tools/linux64/extractor --mimic '/mnt/builds/chromium/src/third_party/llvm-build/Release+Asserts/bin/clang' -MMD -MF obj/v8/d8/d8.o.d -DDCHECK_ALWAYS_ON=1 -DDCHECK_IS_CONFIGURABLE -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_X11=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_GNU_SOURCE '-DCR_CLANG_REVISION="n341867-c2900381-1"' -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -DCOMPONENT_BUILD -D_LIBCPP_ABI_UNSTABLE -D_LIBCPP_ABI_VERSION=Cr -D_LIBCPP_ENABLE_NODISCARD -D_LIBCPP_DEBUG=0 -DCR_LIBCXX_REVISION=375504 -DCR_SYSROOT_HASH=79a7783607a69b6f439add567eb6fcb48877085c -D_DEBUG -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DENABLE_DISASSEMBLER -DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=64 -DENABLE_MINOR_MC -DOBJECT_PRINT -DVERIFY_HEAP -DV8_TRACE_MAPS -DV8_ENABLE_ALLOCATION_TIMEOUT -DV8_ENABLE_FORCE_SLOW_PATH -DV8_ENABLE_DOUBLE_CONST_STORE_CHECK -DV8_INTL_SUPPORT -DENABLE_HANDLE_ZAPPING -DV8_SNAPSHOT_NATIVE_CODE_COUNTERS -DV8_USE_EXTERNAL_STARTUP_DATA -DV8_CONCURRENT_MARKING -DV8_ARRAY_BUFFER_EXTENSION -DV8_ENABLE_LAZY_SOURCE_POSITIONS -DV8_EMBEDDED_BUILTINS -DV8_WIN64_UNWINDING_INFO -DV8_ENABLE_REGEXP_INTERPRETER_THREADED_DISPATCH -DV8_SNAPSHOT_COMPRESSION -DV8_ENABLE_CHECKS -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DV8_TARGET_ARCH_X64 -DV8_HAVE_TARGET_OS -DV8_TARGET_OS_LINUX -DDEBUG -DDISABLE_UNTRUSTED_CODE_MITIGATIONS -DUSING_V8_SHARED -DV8_ENABLE_CHECKS -DV8_COMPRESS_POINTERS -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DUCHAR_TYPE=uint16_t -DUSING_V8_BASE_SHARED -DUSING_V8_PLATFORM_SHARED -I../.. -Igen -I../../v8 -Igen/v8 -I../../v8/include -Igen/v8/include -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../v8/include -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -B../../third_party/binutils/Linux_x64/Release/bin -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -Xclang -mllvm -Xclang -instcombine-lower-dbg-declare=0 -fintegrated-cc1 -fcomplete-member-pointers -m64 -march=x86-64 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -Xclang -fdebug-compilation-dir -Xclang . -no-canonical-prefixes -Wall -Werror -Wextra -Wimplicit-fallthrough -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter '-Wno-c++11-narrowing' -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-range-loop-analysis -fno-omit-frame-pointer -g2 -Xclang -debug-info-kind=constructor -gsplit-dwarf -ggnu-pubnames -ftrivial-auto-var-init=pattern -Xclang -add-plugin -Xclang find-bad-constructs -Xclang -plugin-arg-find-bad-constructs -Xclang check-ipc -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -fno-inline-functions -fno-inline -Wmissing-field-initializers -Wunreachable-code -Wshorten-64-to-32 -O3 -fno-ident -fdata-sections -ffunction-sections -fvisibility=default -Wno-undefined-bool-conversion -Wno-tautological-undefined-compare '-std=c++14' -fno-exceptions -fno-rtti '-nostdinc++' '-isystem../../buildtools/third_party/libc++/trunk/include' '-isystem../../buildtools/third_party/libc++abi/trunk/include' --sysroot=../../build/linux/debian_sid_amd64-sysroot -c ../../v8/src/d8/d8.cc -o obj/v8/d8/d8.o
[E 14:19:38 29432] Is a compiler wrapper; exiting.
```

**Comment by nickrolfe** [comment]
Hey @Manouchehri, you have indeed spotted the problem.

To avoid extracting the same source file twice, we try to detect binaries that are simple wrappers around the real compiler binary, and as part of that there's a special mechanism to distinguish between `clang` and its subprocess `clang -cc1`.

Unfortunately, the new `-fintegrated-cc1` option being used in the V8/Chromium build process throws this mechanism off. The result is that we ignore calls to `clang`, expecting that we will instead intercept a subsequent call to `clang -cc1`, but of course that never happens when `-fintegrated-cc1` is used.

We have already spotted and fixed this problem, but unfortunately it got merged too late for the CodeQL 2.0.6 release. It will be included in the next release, which should happen in around 2 weeks.

In the meantime, if you're somehow able to modify the compiler options to remove `-fintegrated-cc1`, that should let you build a database.

**Comment by Manouchehri** [comment]
Coincidentally I already have a workaround for removing `-fintegrated-cc1` in Chromium. Thanks, I'll give my `clang_use_plugins=false` flag a try and report back!

https://chromium-review.googlesource.com/c/chromium/src/+/2094350
https://github.com/CoatiSoftware/Sourcetrail/issues/852#issuecomment-596253906

**Comment by Manouchehri** [comment]
Yep, looks like my workaround does the trick.

Closing since you already have it fixed for 2.0.7. =) Thanks for the fast response!

**Comment by nickrolfe** [comment]
Great, thanks for letting me know!
